src/Compilers/CSharp/Portable/Symbols/SubstitutedMethodSymbol.cs

```diff
@@ -360,16 +360,24 @@ internal override int CalculateLocalSyntaxOffset(int localPosition, SyntaxTree l
         private int ComputeHashCode()
         {
             int code = this.OriginalDefinition.GetHashCode();
+            int containingHashCode;
 
-            // If the containing type of the original definition is the same as our containing type
-            // it's possible that we will compare equal to the original definition under certain conditions 
-            // (e.g, ignoring nullability) and want to retain the same hashcode. As such, consider only
-            // the original definition for the hashcode when we know equality is possible
-            var containingHashCode = _containingType.GetHashCode();
-            if (containingHashCode == this.OriginalDefinition.ContainingType.GetHashCode() &&
-                wasConstructedForAnnotations(this))
+            if (OriginalDefinition is SynthesizedGlobalMethodSymbol global)
```

Comment:
>if (OriginalDefinition is SynthesizedGlobalMethodSymbol global) [](http://example.com/codeflow?start=12&length=63)

Why wasn't the `else` case appropriate for `SynthesizedGlobalMethodSymbol`?

Reply:
> Why wasn't the else case appropriate for ```SynthesizedGlobalMethodSymbol```?

Containing type for ```SynthesizedGlobalMethodSymbol``` is null.